hummingbot/connector/exchange/chainflip_lp/chainflip_lp_data_formatter.py

```diff
@@ -0,0 +1,282 @@
+        return asset_precision
+        
+    @classmethod
+    def format_orderbook_response(cls,response:Dict, base_asset:Dict[str,str], quote_asset:Dict[str,str]):
+        data = response["result"]
+        bids = list(
+            map(
+                lambda x: {
+                    "amount": cls.format_hex_balance(
+                        x["amount"],
+                        base_asset
+                        ),
+                    "price": cls.format_price(
+                        x["sqrt_price"],
+                        base_asset,
+                        quote_asset,
+                        ),
+                },
+                data["bids"]
+            
+            )
+            )
+        asks = list(
+            map(
+                lambda x: {
+                    "amount": cls.format_hex_balance(
+                        x["amount"],
+                        quote_asset
+                        ),
+                    "price": cls.format_price(
+                        x["sqrt_price"],
+                        base_asset,
+                        quote_asset,
+                    ),
+                },
+                data["asks"]
+            
+            )
+            )
+        format_data = {
+            "bids": bids,
+            "asks": asks
+        }
+        return format_data
+    @classmethod
+    def format_supported_assets(cls, all_pairs:List[Dict[str, str]]):
+        """
+        format the trading pair list of dict to a list of string in
+        the format of {base_asset}-{quote-asset} 
+        """
+        pairs = []
+        for pair in all_pairs:
+            pairs.append(cls.format_assets_to_market_symbol(
+                pair["base_asset"],
+                pair["quote_asset"]
+            ))
+        return pairs
+
+
+    @classmethod
+    def format_trading_pair(cls, pair:str, all_assets:List[Dict[str, str]]):
+        """
+        format a trading pair from {base_asset}-{quote_asset}
+        to the format needed for lp rpc calls
+        e.g ETH-USDT => 
+        {
+            "base_asset":{"chain":"Ethereum","asset":"ETH"},
+            "quote_asset":{"chain":"Ethereum","asset":"USDT"},
+        }
+        """
+        data = {}
+        base_asset, quote_asset = pair.split("-")
+        base_asset_list = list(
+            filter(
+                lambda x: x["asset"].upper() == base_asset.upper(),
+                all_assets
+            )
+        )
+        quote_asset_list = list(
+            filter(
+                lambda x: x["asset"].upper() == quote_asset.upper(),
+                all_assets
+            )
+        )
+        data["base_asset"] = base_asset_list[0] if base_asset_list else None
+        data["quote_asset"] = quote_asset_list[0] if quote_asset_list else None
+
+        return data
+    @classmethod
+    def format_symbol_list(cls, all_assets:List[Dict[str, str]]):
+        """
+        returns a list of just the symbols e.g ["ETH","USDT","BTC"]
```

Comment:
same as above - chainflip supports the same ticker/symbol on multiple chains, eg the `ETH` symbol is available on `Arbitrum` and one `Ethereum`